src/twisted/logger/_observer.py

```diff
@@ -99,24 +55,23 @@ def removeObserver(self, observer):
         except ValueError:
             pass
 
-    def __call__(self, event):
+    def __call__(self, event: LogEvent) -> None:
         """
         Forward events to contained observers.
         """
-        if "log_trace" in event:
+        if "log_trace" not in event:
+            trace = None  # type: Optional[Callable[[ILogObserver], None]]
```

Comment:
Swapped the conditional branches because it's easier to apply the type hint that way.